network.go

```diff
@@ -16,10 +16,27 @@ import (
 )
 
 // download with caching support in the specified cache dir.
-func Download(cacheDir string, url string, progress bool) (string, error) {
+func Download(cacheDir string, url string, progress bool, hash string) (string, error) {
 	name := path.Join(cacheDir, path.Base(url))
 
+	remoteHash, remoteSize, err := getHttpFileInfo(url)
+	if err != nil {
+		// Needed for "working offline"
+		// See https://github.com/anuvu/stacker/issues/44
+		log.Infof("cannot obtain file info of %s", url)
+	}
+	log.Debugf("Remote file: hash: %s length: %s", remoteHash, remoteSize)
+	if hash != "" && hash != remoteHash {
```

Comment:
Should I move this hash check into import.go?

Reply:
Yeah, that might be nice if we can figure out how to do it. Then we'd only have to hash the file once if both hashes were specified.